expat/configure.ac

```diff
@@ -183,33 +183,75 @@ AC_LINK_IFELSE([AC_LANG_SOURCE([
       AC_MSG_RESULT([yes])],
      [AC_MSG_RESULT([no])])])
 
-
-AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
-AC_LINK_IFELSE([AC_LANG_SOURCE([
-  #include <stdlib.h>  /* for NULL */
-  #include <sys/random.h>
-  int main() {
-    return getrandom(NULL, 0U, 0U);
-  }
-  ])],
-  [AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
-   AC_MSG_RESULT([yes])],
-  [AC_MSG_RESULT([no])
-
-   AC_MSG_CHECKING([for syscall SYS_getrandom (Linux 3.17+)])
+AC_ARG_WITH([usegetrandom],
+ [AS_HELP_STRING([--with-usegetrandom],
+ [Use getrandom function in the system @<:@default=check@:>@])],
+ [],
+ [with_usegetrandom=check])
+          
+AS_IF([test "x$with_usegetrandom" != xno],
+  [AC_MSG_CHECKING([for getrandom (Linux 3.17+, glibc 2.25+)])
+   AC_LINK_IFELSE([AC_LANG_SOURCE([
+   #include <stdlib.h>  /* for NULL */
+   #include <sys/random.h>
+   int main() {
+     return getrandom(NULL, 0U, 0U);
+   }
+   ])],
+   [
+	AC_DEFINE([HAVE_GETRANDOM], [1], [Define to 1 if you have the `getrandom' function.])
+    AC_MSG_RESULT([yes])
+    getrandom_detect=yes
+   ],
+    [AC_MSG_RESULT([no])]
+     getrandom_detect=no)
+	],
+   [if test "x$with_usegetrandom" != xcheck; then
+       AC_DEFINE([HAVE_GETRANDOM], [0], [Define to 0, dont use the `getrandom' function.])
+       fi
+   ]) 
+
+if test "x$with_usegetrandom" = xyes; then
+   if test "x$getrandom_detect" = xno; then
```

Comment:
No `AS_IF` here? Could also be combined into one singe `if` but I'd move this whole block into the else branch of `AC_LINK_IFELSE` instead.